### ISSUE TITLE
CKAN crashes if depended mod is not available.

### ISSUE THREAD

**Issue by BiZZKeryear** [KSP-CKAN/CKAN#1514]
__CKAN__ crashes if depended mod is not available (edit: I  now had put the headline in the text since it seems I confused ppl with the rest if I don't)

I run Into this bug while I was trying to update _SCANsat_.

_SCANsat_ says it is depends on _Toolbar_. _Toolbar_ is still marked is incompatible (which isn't ... works fine ... Haven't had a look on the debug console, though ... there might be errors which doesn't affect the game).

When I try to update _SCANsat_ __CKAN__ crashes because it can't install _Toolbar_ (which is btw. already installed ... by __CKAN__ ... when my install was still __KSP 1.0.4__)

**Comment by Olympic1** [comment]
The latest version of toolbar for KSP 1.0.5 is still waiting on a merge.

**Comment by BiZZKeryear** [comment]
But when we put aside the mods involved (only said them so that the bug can be replicated) it still is the issue that the whole CKAN crashes just by selecting a mod which has an unavailable dependency.

Error message ... maybe ... but a crash shouldn't happen

also CKAN should have some more states for mods. Suggestion:
1. __made for version__ XX. Means it works for version XX but most likely not for versions before XX
2. __compatible to version__ YY. Means it also works with YY . YY can be before or after YY.
3. __Barely compatible to version__ YY. Means it works, but it produces Errors in debug console and or or crashes sometimes (in YY). 
4. __unknown__. Means not known yet or not confirmed yet. But if a other mod depends on it, it is probably OK and will be installed after a warning message. If already installed depended mod will also be installed after a warning message.

**Comment by space-is-hard** [comment]
@BiZZKeryear There's a [pull request open](https://github.com/KSP-CKAN/CKAN/milestones/Fuzzy%20versioning) revolving around fuzzy versioning

**Comment by BiZZKeryear** [comment]
@space-is-hard yeah, but this means that the user has to know if or if not a version is compatible.

**Comment by BiZZKeryear** [comment]
Ok, I have final comment ... please have the error messages from #1516 in mind.

I got actually a fix of my own ... and not using CKAN was not an option.
A other mod also showed this behavior (EVE) ... then I had enough
I set CKAN not to update the repositories anymore on start up, was going into the list, edited each entry to to max-version 1.0.5 and since all mods then where "compatible" CKAN didn't had a reason anymore to crash because of incompatible mods, which it wouldn't have touched anyway.

Updated those two mods .. which actually also didn't work since they where kinda linked with two mods which they are now incompatible, to ... so I had all 4 of them uninstall first ... then I was able to reinstall EVE (+ EVE config .. but I count that as one mod)  and SCANsat.
Finally I set the settings back to update on startup and downloaded the correct repository data via refresh button.

Sooo, _**I**_ don't need a fix anymore.
But that still leaves it that there is a nasty bug somewhere and I hope that all information I provided will help to find it.

(CKAN should improve its error catching... a friendly "I am afraid I can't do that, Dave" is always better than a crash message... and being thrown to desktop)

**Comment by Sakata-MC** [comment]
Just a heads-up but this issue is still present.  This is my crash, from selecting to update "Community Resource Pack"
````
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
CKAN.ModuleNotFoundKraken: Cannot install ABLaunchers, module not available
   at CKAN.CkanModule.FromIDandVersion(IRegistryQuerier registry, String mod, KSPVersion ksp_version)
   at CKAN.RelationshipResolver.<RelationshipResolver>c__AnonStorey0.<>m__0(String name)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at CKAN.RelationshipResolver..ctor(IEnumerable`1 module_names, RelationshipResolverOptions options, IRegistryQuerier registry, KSPVersion kspversion)
   at CKAN.MainModList.ComputeConflictsFromModList(IRegistryQuerier registry, IEnumerable`1 change_set, KSPVersion ksp_version)
   at CKAN.Main.<UpdateChangeSetAndConflicts>c__async1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CKAN.Main.<ModList_CellValueChanged>c__async0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1078.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
ckan
    Assembly Version: 0.0.0.0
    Win32 Version: 0.0.0.0
    CodeBase: file:///C:/Games/Steam/steamapps/common/Kerbal%20Space%20Program/ckan.exe
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1075.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1064.2 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1078.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.6.1064.2 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Microsoft.CSharp
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.CSharp/v4.0_4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------
System.Dynamic
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Dynamic/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Dynamic.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Assembly Version: 0.0.0.0
    Win32 Version: 4.6.1078.0 built by: NETFXREL3STAGE
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/mscorlib/v4.0_4.0.0.0__b77a5c561934e089/mscorlib.dll
----------------------------------------
System.Transactions
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.Transactions/v4.0_4.0.0.0__b77a5c561934e089/System.Transactions.dll
----------------------------------------
System.Runtime.Serialization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Serialization/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Serialization.dll
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.EnterpriseServices
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.EnterpriseServices/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.EnterpriseServices.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.6.1038.0 built by: NETFXREL2
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
````

**Comment by nobodyhasthis** [comment]
If that last reported error by @Sakata-MC was from an update of "Community Resource Pack"
The actual message thrown looks related to "ABLaunchers".

@politas sorry if I am wrong but this looks like another version of #1686 ?

I am asking because a problem that I had last week was fixed. However the unusual program behaviour was interesting enough that I was told to raise a new issue to report it. Looking at message here it looks very similar. So cross referencing in the hope that it helps. If it does not help please let me know and I will delete this message.
